### ISSUE TITLE
Can I test it with the ssd tensorflow lite model I trained?

### ISSUE THREAD

**Issue by SunMars** [asus4/tf-lite-unity-sample#121]
Thank you for your work. Can I test it with the ssd tensorflow lite model I trained? Mobilenet has a resolution of 320x320. If it can be used, where do I need to adjust the parameters?

**Comment by asus4** [comment]
Please note that the SSD model in this repo is: 
- Quantized to the integer 8bit. some models might have 16bit float input.
- Contains a custom ops - NMS: Non Max Suppression 

If your model is in a different format, you will need to modify the code slightly.

**Comment by yokewang** [comment]
Since this great work is a unity porting of  "TensorFlow Lite Examples",  will it support 16bit float model in the future?

**Comment by asus4** [comment]
@yokewang I think you can get it working with a few tweaks.

If the only difference between the models is `float`, simply change `sbyte` to `float`.

https://github.com/asus4/tf-lite-unity-sample/blob/2186253231f7062b63b71a4eda363dd0ba3bb063/Assets/Samples/SSD/SSD.cs#L5

**Comment by SunMars** [comment]
We quantified the model as you said. But still wrong.The parameters of the model are shown in the figure：can you point out the problem?
https://cyccis888.oss-cn-shanghai.aliyuncs.com/test/%E6%88%AA%E5%B1%8F2021-04-28%20%E4%B8%8B%E5%8D%885.04.03.png

Unity console ：
> Exception: Failed to create TensorFlowLite Interpreter
TensorFlowLite.Interpreter..ctor (System.Byte[] modelData, TensorFlowLite.InterpreterOptions options) (at Packages/com.github.asus4.tflite/Runtime/Interpreter.cs:67)
TensorFlowLite.BaseImagePredictor`1[T]..ctor (System.String modelPath, System.Boolean useGPU) (at Assets/Samples/Common/BaseImagePredictor.cs:49)
TensorFlowLite.SSD..ctor (System.String modelPath) (at Assets/Samples/SSD/SSD.cs:22)
SsdSample.Start () (at Assets/Samples/SSD/SsdSample.cs:27)

**Comment by Sacristan** [comment]
@asus4 Hey! On related note - do You have some sort of notes how to build custom model ? 

Since all my models throw 
Exception: Failed to create TensorFlowLite Interpreter

Is there a way how to get failure code so i can check what exactly is wrong ?

**Comment by asus4** [comment]
@SunMars @Sacristan sorry for the late reply.
 
These error messages were not enough to tell me what the problem was. Please understand that this repository only provides samples for google's pre-trained model as it is difficult to make samples for a variety of custom models.

**Comment by asus4** [comment]
I'm busy at the moment, but there seems to be a lot of interest in custom SSD models, so I'm hoping to add the EfficientDet example made by [model_maker](https://www.tensorflow.org/lite/tutorials/model_maker_object_detection) when I have time.

**Comment by manena** [comment]
Hello, this project is amazing! Thanks a lot for sharing!

I started using it for audio command recognition a few months ago and now I want to use my own trained models. I trained some 1 second commands, but it seems my tflite model doesn’t have the expected properties.

I get lost with properties and dimensions and would like to have a guide to adapt my model to what the system expects… for example, I’m not sure if just by adapting input and output properties will do the trick.

thanks for your help,
Manena

**Comment by asus4** [comment]
Hi @manena, Thanks for using my repo.

I am also interested in audio recognition but haven't any experience with it at the moment. You may find similar posts on the [TensorFlow Forum](https://discuss.tensorflow.org/) or TensorFlow GitHub.

**Comment by manena** [comment]
Hello, thanks for the link!
After sending the message, I remembered that audio recognition was not included in the project and that I adapted it months ago to use a voice command model I found online.

Fortunately, I’ve found out what is wrong when I try to use my own tensorflow model: the model expects a spectrogram, and I’m trying to feed it with a wave file.

I will adapt the model to receive a wav and then see if it’s correct.

All this is so interesting!

**Comment by NeilEntote** [comment]
Hi I've been trying to get the SSD code to work for my own custom model. I've determined that its quantized to float32, so I've changed sbyte to float as you recommended. However, I'm not sure how to determine if mine has/does not have NMS. Is there anything in the code that needs to be changed for this?

Also, my model has a different input shape than the default, and I was wondering how to change this in the code, as in Issue #89   you mentioned this, but I haven't been able to figure out what to change. 
![image](https://user-images.githubusercontent.com/56211468/125839897-5dd905b0-e379-41f2-a902-1316c2488dda.png)

The specific error I've been getting is "Failed to create TensorflowLite Model" from line 61 of packages/com.github.asus4.tflite/Runtime/interpreter.cs
![image](https://user-images.githubusercontent.com/56211468/125840181-abd74454-f1f7-47e2-9710-18ba0f25baed.png)

Model Properties:
![image](https://user-images.githubusercontent.com/56211468/125840304-81bb5950-6fb9-482c-b5c0-405c3b569d9b.png)

Input:
![image](https://user-images.githubusercontent.com/56211468/125840335-fc8f2198-c9e8-45eb-9084-d63c3aa38c3e.png)

Thanks!
